### PR TITLE
Display outliers in heatmaps

### LINKED ISSUE #916: [Panorama] Display outliers in heatmaps
### Tasks
- [x] Create a new java action to return outlier counts by peptide 
- [x] Create an html page with date range selector and calling the above action to get the data
- [x] Create outlier heatmap table with values having color gradient
- [x] Create a new webpart for the UI
- [x] Pass the date range to the action and filter the data 
- [x] Persist and make date range filter in synch with qc plots dashboard 📌 
- [x] Display total number of replicates and total outliers
- [x] code review
- [x] Test plan @ankurjuneja 
- [x] Manual testing @labkey-sweta 
- [x] verify TC & merge

### PR DESCRIPTION
**Pull request by ankurjuneja**
#### Rationale
A heatmap is a valuable tool to assist Panorama users in identifying patterns of outlier data, and help them to make better decisions regarding their data and experiments. In this PR, a heatmap view for outliers in QC folders is implemened.

#### Changes
* new webpart for the outlier heatmap summary

### CODE REVIEW

File: resources/views/outlierHeatmapSummary.html [hunk 1]
```diff
@@ -0,0 +1,401 @@
...
+        return maxDate;
+    }
+
+    function generateTable(data) {
+        const tableContainer = document.getElementById('table-container');
+        let tableHTML = '<table id="heatmap-table" class="heatmap-table">';
+        tableHTML += '<thead><tr><th></th>';
+
+        for (let i = 0; i < data.metrics.length; i++) {
+            tableHTML += '<th>' + data.metrics[i] + '</th>';
```
Comment (labkey-jeckels) [inline resources/views/outlierHeatmapSummary.html]: HTML encoding

File: resources/views/outlierHeatmapSummary.html [hunk 2]
```diff
@@ -0,0 +1,401 @@
...
+        }
+        tableHTML += '<th><b>Total</b></th>';
+
+        tableHTML += '</tr></thead><tbody>';
+
+        let totalOutliersByMetric = {};
+        for (let i = 0; i < data.peptideOutliers.length; i++) {
+            tableHTML += '<tr><td>' + data.peptideOutliers[i].peptide + '</td>';
```
Comment (labkey-jeckels) [inline resources/views/outlierHeatmapSummary.html]: HTML encoding

File: resources/views/outlierHeatmapSummary.html [hunk 3]
```diff
@@ -0,0 +1,401 @@
...
+    function getData() {
+        LABKEY.Ajax.request({
+            url: LABKEY.ActionURL.buildURL('targetedms', 'GetPeptideOutliers.api'),
+            params: {
+                startDate: startDate,
+                endDate: endDate
+            },
+            success: function (response) {
+                const parsed = JSON.parse(response.responseText);
+                document.getElementById('total-replicates').innerHTML = '&nbsp' +  parsed.replicatesCount;
```
Comment (labkey-jeckels) [inline resources/views/outlierHeatmapSummary.html]:
```suggestion
                document.getElementById('total-replicates').innerHTML = '&nbsp;' +  parsed.replicatesCount;
```

File: resources/views/outlierHeatmapSummary.html [hunk 4]
```diff
@@ -0,0 +1,401 @@
...
+
+    function customDateRange() {
+        if (document.getElementById("start-date").value) {
+            startDate = formatDate(document.getElementById("start-date").value);
+        }
+        if (document.getElementById("end-date").value) {
+            endDate = formatDate(document.getElementById("end-date").value);
+        }
+        if (document.getElementById("start-date").value && document.getElementById("end-date").value) {
+            console.log("Custom date range selected " + startDate + " " + endDate);
```
Comment (labkey-jeckels) [inline resources/views/outlierHeatmapSummary.html]:
```suggestion
```

File: resources/views/outlierHeatmapSummary.html [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+                    maxDate = data.rows[0]['MaxAcquiredTime'] ? new Date(data.rows[0]['MaxAcquiredTime']) : null;
+                    if (dateRangeOffset !== -1) {
+                        startDate = calculateStartDateByOffset(dateRangeOffset);
+                        endDate = calculateEndDateByOffset(dateRangeOffset);
+                    }
+                    getData();
+                }
+            },
+            failure: function(response) {
+                Ext4.get(plotPanelId).update("<span class='labkey-error'>Error: " + response.exception + "</span>");
```
Comment (labkey-jeckels) [inline resources/views/outlierHeatmapSummary.html]: HTML encode

**Review by labkey-jeckels — APPROVED** [review]
